schemas.sql

```diff
@@ -0,0 +1,112 @@
+
+-- keep track of all users (id, tag, name)
+
+CREATE TABLE users (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR NOT NULL,
+    user_tag VARCHAR NOT NULL,
+    user_name VARCHAR NOT NULL,
+    deleted_at TIMESTAMP
+);
+
+-- From rememberMessage.js
+-- 'content' is a VARCHAR and not TEXT because 
+-- discord has a limit of 2000 chars per message (4000 for nitro users)
+-- and VARCHAR has a limit of 10485760
+CREATE TABLE messages (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR NOT NULL,
+    content VARCHAR NOT NULL,
+    message_id VARCHAR NOT NULL,
+    message_ts VARCHAR NOT NULL,
+    deleted_at TIMESTAMP
+);
+
+-- keep in mind, array size is just verbose and effectively does nothing
+CREATE TABLE checkin_schedules (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR NOT NULL,
+    utc_days VARCHAR[],
+    utc_time INT[2],
+    local_days VARCHAR[],
+    local_time INT[2],
+    deleted_at TIMESTAMP
+);
+
+-- -
+-- -
+-- - I HAVE NOTHING FOR AVAILABILITY YET
+-- -
+-- -
+
+-- from/to are dayjs strings. Ex: `JSON.stringify(dayjs())`
+CREATE TABLE unavailable_schedules (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR NOT NULL,
+    from_time VARCHAR,
+    to_time VARCHAR,
+    reason VARCHAR,
+    deleted_at TIMESTAMP
+);
+
+-- this is not yet determined
+-- instead of payload, consider using regular entries
+CREATE TABLE checkin_queue
+(
+    checkin_q_pk UUID PRIMARY KEY,
+    time_inserted TIMESTAMP,
+    payload JSON,
+);
+
+CREATE INDEX checkin_q_time_inserted_id
+    ON checkin_queue (time_inserted ASC);
+
+-- this is for rose thorn bud responses
+CREATE TABLE checkin_responses (
+    id SERIAL PRIMARY KEY,
+    user_id VARCHAR NOT NULL,
+    rose VARCHAR,
+    thorn VARCHAR,
+    bud VARCHAR,
```

Comment:
See conversation in Discord about saving rose, thorn, and bud in a JSON field instead.